### PR TITLE
Phpcstask fileset support

### PR DESCRIPTION
**Pull request by kenguest**
Add support for filesets to the phpcs (version 3) task.

### CODE REVIEW

File: classes/phing/tasks/ext/PhpCSTask.php [hunk 1]
```diff
@@ -103,9 +112,26 @@ public function setFile(PhingFile $file): void
...
+        if ($this->file === null && count($this->filesets) == 0) {
+            throw new BuildException('Missing both attribute "file" and "fileset".');
+        }
+        if ($this->file === null) {
+            // check filesets, and compile a list of files for phpcs to analyse
+            foreach ($this->filesets as $fileset) {
+                $scanner = $fileset->getDirectoryScanner($project);
+                $files = $scanner->getIncludedFiles();
+                $fromDir = $fileset->getDir($project);
+                foreach ($files as $file) {
```
Comment (siad007) [inline classes/phing/tasks/ext/PhpCSTask.php:130]: Is it possible to use the iterator here?

File: test/classes/phing/types/PatternSetTest.php [hunk 2]
```diff
@@ -1,83 +0,0 @@
...
- * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
- * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- *
- * This software consists of voluntary contributions made by many individuals
- * and is licensed under the LGPL. For more information please see
- * <http://phing.info>.
- */
-
-class PatternSetTest extends \PHPUnit\Framework\TestCase
```
Comment (siad007) [inline test/classes/phing/types/PatternSetTest.php:20]: Deleted?

Comment (kenguest) [reply test/classes/phing/types/PatternSetTest.php:20]: Removed in commit  56a86ba0da7fd7a90da6a1fcf1452c2e0fc5080d by Thomas Müller <mimmi20@live.de> on Nov 16th 2019.

Comment (siad007) [reply test/classes/phing/types/PatternSetTest.php:20]: Ah ok - will check.

File: classes/phing/tasks/ext/PhpCSTask.php [hunk 3]
```diff
@@ -34,6 +34,15 @@ class PhpCSTask extends Task
      */
     private $file;
 
+    /**
+     * The
+     *
+     * @var array
+     */
+    protected $filesets = [];
```
[the remark this answers is not in the document]
Comment (kenguest) [reply classes/phing/tasks/ext/PhpCSTask.php]: I've switched over to the trait now, in changeset a4f34801cceb009da33e7869530571efa07d089c

File: classes/phing/tasks/ext/PhpCSTask.php (hunk 1 above)
Comment (kenguest) [reply classes/phing/tasks/ext/PhpCSTask.php:127]: I've switched over to using getIterator now, in changeset a4f34801cceb009da33e7869530571efa07d089c

File: test/classes/phing/types/PatternSetTest.php (hunk 2 above)
Comment (kenguest) [reply test/classes/phing/types/PatternSetTest.php:20]: Thanks.

**Comment by siad007** [comment]
@kenguest thx looks good - is merged

**Pull request merged**